# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/cookiestore/CookieStore.h
# Commit: cc3c39aa8fb8
# Full Hash: cc3c39aa8fb8124cca265e84f71a29cbd218b704
# Author: Andrea Marchesini <amarchesini@mozilla.com>
# Date: 2024-09-12 09:23:07
# Regressor Bug: 1475599
# File Overlap Count: 1
# Description:
#   Bug 1475599 - part 2 - CookieStore API - IPDL, r=edgul,webidl,smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D215145
# ==============================================================================

diff -r 2fc9a3d0ad8b -r cc3c39aa8fb8 dom/cookiestore/CookieStore.h
--- a/dom/cookiestore/CookieStore.h	Wed Sep 11 16:26:54 2024 +0000
+++ b/dom/cookiestore/CookieStore.h	Wed Sep 11 16:26:55 2024 +0000
@@ -15,6 +15,9 @@
 
 namespace mozilla::dom {
 
+class CookieData;
+class CookieStoreChild;
+class CookieStoreNotifier;
 class Promise;
 
 class CookieStore final : public DOMEventTargetHelper {
@@ -29,6 +32,8 @@
   JSObject* WrapObject(JSContext* aCx,
                        JS::Handle<JSObject*> aGivenProto) override;
 
+  void FireDelayedDOMEvents();
+
   already_AddRefed<Promise> Get(const nsAString& aName, ErrorResult& aRv);
 
   already_AddRefed<Promise> Get(const CookieStoreGetOptions& aOptions,
@@ -54,6 +59,19 @@
  private:
   explicit CookieStore(nsIGlobalObject* aGlobal);
   ~CookieStore();
+
+  void Shutdown();
+
+  Document* MaybeGetDocument() const;
+
+  already_AddRefed<Promise> GetInternal(const CookieStoreGetOptions& aOptions,
+                                        bool aOnlyTheFirstMatch,
+                                        ErrorResult& aRv);
+
+  bool MaybeCreateActor();
+
+  RefPtr<CookieStoreChild> mActor;
+  RefPtr<CookieStoreNotifier> mNotifier;
 };
 
 }  // namespace mozilla::dom